botcore/_bot.py

```diff
@@ -0,0 +1,265 @@
+import asyncio
+import socket
+import types
+from abc import abstractmethod
+from contextlib import suppress
+from typing import Optional
+
+import aiohttp
+import discord
+from async_rediscache import RedisSession
+from discord.ext import commands
+
+from botcore.async_stats import AsyncStatsClient
+from botcore.site_api import APIClient
+from botcore.utils._extensions import walk_extensions
+from botcore.utils.logging import get_logger
+
+log = get_logger()
+
+
+class StartupError(Exception):
+    """Exception class for startup errors."""
+
+    def __init__(self, base: Exception):
+        super().__init__()
+        self.exception = base
+
+
+class BotBase(commands.Bot):
+    """A sub-class that implements many common features that Python Discord bots use."""
+
+    def __init__(
+        self,
+        *args,
+        guild_id: int,
+        prefix: str,
+        allowed_roles: list,
+        intents: discord.Intents,
```

Comment:
Aren't these redundant? Just rely on passing them via `*args` or `**kwargs`.

Reply:
Yea, good point. I'm tempted to leave `allowed_roles` in there though, to ensure it's required to be used.

Reply:
https://github.com/python-discord/bot-core/pull/42/commits/90796054965b4ed44fce636c7b64181958357574